portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,20 +30,47 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> comments;
+  class Comment {
+      private String name;
+      private String comment; 
+      private long timestamp;
 
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
+      public Comment(String name, String comment, long timestamp) {
+          this.name = name;
+          this.comment = comment; 
+          this.timestamp = timestamp;
+      }
   }
 
+  private List<Comment> comments;
+
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    comments = new ArrayList<>();
+    
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int numComments = getLimit(request);
+
+    for (Entity entity : results.asIterable()) {
```

Comment:
Were you able to add the limit to the query, so you don't end up pulling all the comments into memory?